### ISSUE TITLE
Fatal run-time error when going live (works on local dev)

### ISSUE THREAD

**Issue by git-johan** [nfourtythree/craft-json-data#2]
Hi,

great addon and does exactly what I want it to do on my local environment. When I upload everything to the live server on the other hand, i get this error message:

```
Type: Fatal run-time error
Message: Uncaught Error: Class 'Craft\JsonDataTwigExtension' not found in /var/www/vhosts/brattvag-il.no/craft/plugins/jsondata/JsonDataPlugin.php:30
Stack trace:
#0 [internal function]: Craft\JsonDataPlugin->addTwigExtension()
#1 /var/www/vhosts/brattvag-il.no/craft/app/services/PluginsService.php(582): call_user_func_array(Array, Array)
#2 /var/www/vhosts/brattvag-il.no/craft/app/services/TemplatesService.php(1293): Craft\PluginsService->call('addTwigExtensio...')
#3 /var/www/vhosts/brattvag-il.no/craft/app/services/TemplatesService.php(162): Craft\TemplatesService->_addPluginTwigExtensions(Object(Craft\TwigEnvironment))
#4 /var/www/vhosts/brattvag-il.no/craft/app/services/TemplatesService.php(221): Craft\TemplatesService->getTwig()
#5 /var/www/vhosts/brattvag-il.no/craft/app/controllers/BaseController.php(74): Craft\TemplatesService->render('500', Array)
#6 /var/www/vhosts/brattvag-il.no/craft/app/controllers/TemplatesController.php(201): Craft\BaseController->renderTemplate('500', Array)
#7 /var/www/vhosts/brattva
File: /var/www/vhosts/brattvag-il.no/craft/plugins/jsondata/JsonDataPlugin.php
Line: 30
URL: /admin/settings/plugins
GET: a:1:{s:1:"p";s:22:"admin/settings/plugins";}
POST: a:0:{}
IP: 62.92.231.134

Array
(
    [0] => Array
        (
            [function] => shutdown_function
            [args] => Array
                (
                )

        )

)
```

Do you have any idea what might cause it? When I deactivate / remove the addon, everything works just fine.

**Comment by nfourtythree** [comment]
Hi

Just wondering what version of Craft are you using? Also is it the same version on local and on the live?

**Comment by git-johan** [comment]
Hi, I use version Craft Pro 2.5.2761. Same on both.

**Comment by nfourtythree** [comment]
Have just tested this and things seem to look alright.

Sorry for the further questions:
Is it definitely installed?
What version of PHP is on the live server?

Just trying to get to the bottom of this for you!

**Comment by git-johan** [comment]
No problem.

Yes, it's definitely installed. I have tried both PHP version 7.0.2 and 5.6.17.

**Comment by git-johan** [comment]
The problem occurs when I install / activate it.

**Comment by nfourtythree** [comment]
This is an interesting one! Have tried it on both the PHP versions you have stated and I still can't break it!

Two more questions:
1. Is the folder name for the plugin lower case?
2. In the file `plugins/jsondata/JsonDataPlugin.php` try changing line 28 to the following: `Craft::import('plugins.jsondata.twigextensions.JsonDataTwigExtension');`

Basically it seems like it isn't loading the twig extension but only on the live server so makes me think for some reason it is struggling to find the file

**Comment by git-johan** [comment]
1. Yes, the folder is in lowercase. I just re-installed it and forgot to change the foldername. The error message I then got was easy to understand. Craft couldn't see the plugin under Settings > Plugins. Changed the name, plugin popped up under settings, installed it and site broke.
2. ) Started answering before testing. Now it works. Either changing the line was the fix, or there were some deploy cache bug on the server. Any way, it's all good. Thanks!
